### PR TITLE
WL-3959 HTML Entities for quotes 

### PR DESCRIPTION
**Pull request by nicholaswilson100**
Also: 

Javascript confirm pop up for Delete
Fix bug when manually creating citation where it re-adds nested citations

### CODE REVIEW

File: citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java [hunk 1]
```diff
@@ -1083,9 +1084,10 @@ protected void doPostJsonResponse(ParameterParser params,
 			String addSectionHTML = params.getString("addSectionHTML");
 			int locationId = params.getInt("locationId");
 			String cleanAddSectionHTML = getFormattedText().processFormattedText(addSectionHTML, new StringBuilder(), true, true);
+			cleanAddSectionHTML = cleanAddSectionHTML.replaceAll("'", "&#8217;").replaceAll("\"", "&#8220;");
```
Comment (buckett) [inline citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java]: Why are you doing this?

File: citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java [hunk 2]
```diff
@@ -3273,7 +3276,7 @@ public void doImport ( RunData data)
 		}
 
 		CitationCollection collection = null;
-        collection = getCitationCollection(state, false);
+        collection = getCitationService().getUnnestedCitationCollection(citationCollectionId);
```
Comment (buckett) [inline citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java]: Indentation.

File: citations-tool/tool/src/webapp/js/edit_nested_citations.js [hunk 3]
```diff
@@ -319,13 +319,15 @@
         }
 
         function removeSection() {
-            var actionUrl = $('#newCitationListForm').attr('action');
-            $('#citation_action').val('remove_section');
-            var params = $('#newCitationListForm').serializeArray();
-            params.push({name:'locationId', value:this.id.replace('removeSection', '')});
-            ajaxPost(actionUrl, params, false);
+            if(confirm("Are you sure you want to delete this section (and any subsections or citations it has)?")) {
```
Comment (buckett) [inline citations-tool/tool/src/webapp/js/edit_nested_citations.js]: i18n on the confirmation.

File: citations-tool/tool/src/webapp/vm/citation/_listNestedCitations.vm [hunk 4]
```diff
@@ -19,7 +19,7 @@
 		<li id='$linkId' class='months h1Editor' data-value='$nestedSection.getValue()' data-location='$nestedSection.getLocation()'
 			data-sectiontype='$nestedSection.getSectiontype()'>
 			<div id='$editorDivId' class='editor h1Editor'>
-				$nestedSection.getValue()
+				#if($nestedSection.getValue())$nestedSection.getValue()#end
```
Comment (buckett) [inline citations-tool/tool/src/webapp/vm/citation/_listNestedCitations.vm]: In velocity you can use `$!nestedSection.value` which will only output the value if it's defined.
http://velocity.apache.org/engine/devel/user-guide.html#quietreferencenotation
Don't have to change it but it's much cleaner.

File: citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java (hunk 1 above)
Comment (nicholaswilson100) [inline citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java]: The reason I'm replacing single and double quotes with their HTML entities is that they are used in the attributes of <li> tags and quotes screw up the HTML.  Escaping them by usual means (e.g., \' or \") does not work in HTML:  

http://stackoverflow.com/questions/2428572/how-to-escape-single-quote

I'll put a comment in the code.

**Comment by nicholaswilson100** [comment]
I've made various changes. 

i18n ; indentation ; html entity explanation

**Pull request merged**